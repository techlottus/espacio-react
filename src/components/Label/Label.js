import React, { useEffect } from 'react'

export const Label = React.memo(({data, onClick}) => {

  const labelRef = React.createRef();

  useEffect(() => {
    labelRef.current.data = {
        id: data.url || '',
        type: data.type || '',
        title: data.title || '',
        icon: data.icon || '',
        iconEvent: data.iconEvent || '',
        size: data.size || '',
        disabled: data.disabled || false,
    };
  },[data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    labelRef.current.addEventListener('onClick', onClick)
  },[]) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-label ref={labelRef}></lottus-label>
    </>
  )
});

