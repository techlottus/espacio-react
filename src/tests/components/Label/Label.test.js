import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Label } from "../../../components/Label/Label";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState;
let data = {
    id: "",
    type: "",
    title: "",
    icon: "",
    iconEvent: "",
    size: "",
    disabled: false,
}

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Label data={data} />
    </Provider>
  );
};

describe("Test <Label/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
