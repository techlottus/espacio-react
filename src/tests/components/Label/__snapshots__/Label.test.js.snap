// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Label/> should init component 1`] = `<lottus-label />`;
